tools/azsdk-cli/ci.yml

```diff
@@ -27,6 +27,7 @@ extends:
     ToolDirectory: tools/azsdk-cli/Azure.Sdk.Tools.Cli
     ReleaseBinaries: true
     TestDirectory: tools/azsdk-cli/Azure.Sdk.Tools.Cli.Tests
+    PublishEnvironment: 'tools'  # require engsys team approval for release
```

Comment:
Let's use the general 'publish-package" environment we don't need to create another for just this.

Reply:
@weshaggard `publish-package` is permissioned for azure sdk partners. I assumed we wanted publish permissions for this package to be a bit more limited?